database/pgx/pgx.go

```diff
@@ -0,0 +1,442 @@
+// +build go1.9
+
+package pgx
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"io/ioutil"
+	nurl "net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/golang-migrate/migrate/v4/database/multistmt"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx"
+	_ "github.com/jackc/pgx/stdlib"
+)
+
+func init() {
+	db := Postgres{}
+	database.Register("pgx", &db)
+}
+
+var (
+	multiStmtDelimiter = []byte(";")
+
+	DefaultMigrationsTable       = "schema_migrations"
+	DefaultMultiStatementMaxSize = 10 * 1 << 20 // 10 MB
+)
+
+var (
+	ErrNilConfig      = fmt.Errorf("no config")
+	ErrNoDatabaseName = fmt.Errorf("no database name")
+	ErrNoSchema       = fmt.Errorf("no schema")
+	ErrDatabaseDirty  = fmt.Errorf("database is dirty")
+)
+
+type Config struct {
+	MigrationsTable       string
+	DatabaseName          string
+	SchemaName            string
+	StatementTimeout      time.Duration
+	MultiStatementEnabled bool
+	MultiStatementMaxSize int
+}
+
+type Postgres struct {
```

Comment:
Thoughts about renaming this to `Pgx` or keeping `Postgres`? Wondering if `pgx.Pgx` stutters or if `pgx.Postgres` is confusing.

Reply:
I left it as is simply because it is ultimately a Postgres driver, but I defer to your judgement if it's confusing or not sufficiently distinguished from the existing Postgres driver.

Reply:
Let's keep `pgx.Postgres`